bitbots_utils/bitbots_utils/game_settings.py

```diff
@@ -29,88 +40,92 @@ def provide_config(path):
+    print("=============== {} ===============".format(name))
+    if valid_options:
+        print(f"Options: {valid_options}")
+    print(f"Explanations: {explanation}")
+
     if current_value is not None:
-        input_prompt = 'Value ({}): '.format(current_value)
+        input_prompt = f"Value ({current_value}): "
     else:
-        input_prompt = 'Value: '
+        input_prompt = "Value: "
 
     value_is_valid = False
     while not value_is_valid:
         new_value = input(input_prompt).lower()
 
-        if new_value == '':
+        if new_value == "":
             if current_value is not None:
                 new_value = current_value
                 value_is_valid = True
         else:
-            value_is_valid = check_new_value(new_value, definition)
-    def_type = type(definition[0])
-    return def_type(new_value)
+            value_is_valid = check_new_value(new_value, value_type, valid_options)
+
+    return value_type(new_value)
 
 
-def check_new_value(new_value: str, definition) -> bool:
+def check_new_value(new_value: str, value_type: Any, valid_options: Optional[list[Any]] = None) -> bool:
     """
     checks with definition if new value is a valid input
     :param new_value: input to set as new value
-    :param definition: valid options for new value
+    :param value_type: type object of the type of value to cast to
+    :param valid_options: valid options for new value
     :return: true if valid, false if not
     """
 
-    
-
-    definitiontype = type(definition[0])
-
     try:
-        new_value = definitiontype(new_value) # casts value to the type of
-    except:
-        print("{} could not be converted to a {}. Are you sure it is in the right format?".format(new_value,definitiontype))
-
+        new_value = value_type(new_value)
+    except Exception:
+        print(f"{new_value} could not be converted to a {value_type}. Are you sure it is in the right format?")
+        return False
 
-    if new_value in definition:
-        return True
-    else:
-        # print(new_value, definition)
-        print(' {} no valid option'.format(new_value))
+    not_a_valid_option = valid_options is not None and new_value not in valid_options
+    if not_a_valid_option:
+        print(f" {new_value} no valid option")
         return False
 
+    return True
 
 
 def main():
     config = provide_config(SETTING_PATH)
-    #config = config['parameter_blackboard']['ros_parameters']
-    ros_parameters = config['parameter_blackboard']['ros__parameters']
+    ros_parameters = config["parameter_blackboard"]["ros__parameters"]
     if ros_parameters is None:
         ros_parameters = {}
-        config['parameter_blackboard']['ros__parameters'] = ros_parameters
-        
+        config["parameter_blackboard"]["ros__parameters"] = ros_parameters
+
     options = provide_config(OPTIONS_PATH)
-    
     for key in options.keys():
+        entry_value_type = locate(options[key]["type"])
```

Comment:
What is pydoc.locate?

Reply:
It can import stuff from a string, i.e. `ipaddress.IPv4Address`.